### PR TITLE
Azure EventHub - Honour timeout in Health Check

### PR DESCRIPTION
**Pull request by cieciurm**
**What this PR does / why we need it**:
This PR adds honouring of timeout from `HealthCheckContext` to creation of `EventHubProducerClient`.

**Which issue(s) this PR fixes**: #1786

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: No.

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: build/versions.props [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <HealthCheckAzureApplicationInsights>7.0.0</HealthCheckAzureApplicationInsights>
     <HealthCheckAzureDigitalTwin>7.0.0</HealthCheckAzureDigitalTwin>
     <HealthCheckAzureSearch>7.0.0</HealthCheckAzureSearch>
-    <HealthCheckAzureServiceBus>7.0.0</HealthCheckAzureServiceBus>
+    <HealthCheckAzureServiceBus>7.1.0</HealthCheckAzureServiceBus>
```
Comment (sungam3r) [inline build/versions.props:14]: Please merge `master` to reduce diff here.

File: src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ namespace HealthChecks.AzureServiceBus;
 public class AzureEventHubHealthCheck : IHealthCheck
 {
     private const string ENTITY_PATH_SEGMENT = "EntityPath=";
+    private const int DEFAULT_TIME_OUT_IN_SECONDS = 60;
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs:10]:
```suggestion
    private const int DEFAULT_TIMEOUT_IN_SECONDS = 60;
```

File: src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs [hunk 3]
```diff
@@ -78,4 +81,14 @@ private string GetFullConnectionString()
...
         return connectionString;
     }
+
+    private static EventHubProducerClientOptions CreateClientOptions(HealthCheckContext context) => new()
+    {
+        ConnectionOptions = new()
+        {
+            ConnectionIdleTimeout = context.Registration.Timeout.TotalMilliseconds > 0
+                ? context.Registration.Timeout
+                : TimeSpan.FromSeconds(DEFAULT_TIME_OUT_IN_SECONDS), // Default in EventHub SDK
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs:91]: I would not set default value. Just do not set value at all.

Comment (cieciurm) [reply src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs]: Thanks for suggestion, it's been changed.

**Comment by sungam3r** [comment]
Thanks.

**Pull request merged**